packages/myst-to-tex/src/index.ts

```diff
@@ -301,13 +363,68 @@ const handlers: Record<string, Handler> = {
   },
 };
 
+class TexGlossarySerializer {
+  static COMMENT_LENGTH = 50;
+
+  preambleGlossary: string;
+  printedGlossary: string;
+
+  constructor(glossaryDefinitions: Record<string, [DefinitionTerm, DefinitionDescription]>) {
+    this.printedGlossary = this.renderGlossary();
+    this.preambleGlossary = this.renderGlossaryImports(glossaryDefinitions);
+  }
+
+  private renderGlossary(): string {
+    const block = writeTexLabelledComment(
+      'glossary',
+      ['\\printglossaries'],
+      TexGlossarySerializer.COMMENT_LENGTH,
+    );
+    const percents = ''.padEnd(TexGlossarySerializer.COMMENT_LENGTH, '%');
+    return `${percents}\n${block}${percents}`;
+  }
+
+  private renderGlossaryImports(
+    directives: Record<string, [DefinitionTerm, DefinitionDescription]>,
+  ): string {
+    if (!directives) return '';
+    const block = writeTexLabelledComment(
+      'glossary',
+      this.createGlossaryDirectives(directives),
+      TexGlossarySerializer.COMMENT_LENGTH,
+    );
+    if (!block) return '';
+    const percents = ''.padEnd(TexGlossarySerializer.COMMENT_LENGTH, '%');
+    return `${percents}\n${block}${percents}`;
+  }
+
+  private createGlossaryDirectives(
+    glossaryDefinitions: Record<string, [DefinitionTerm, DefinitionDescription]>,
+  ): string[] {
+    const directives = Object.keys(glossaryDefinitions).map((k) => ({
+      key: k,
+      name: (glossaryDefinitions[k][0].children[0] as GenericNode).value || '',
+      description: (glossaryDefinitions[k][1].children[0] as GenericNode).value || '',
```

Comment:
The values for `name` and `description` here need to be computed differently. These assume the `DefinitionTerm` and `DefinitionDescription` nodes each only have a single `text` child. However, if that's not the case it will break (I think this is a pretty realistic scenario, e.g. emphasis in a definition description).

The simplest option is to use `toText(glossaryDefinitions[k][1])` from `myst-common` - this will flatten all the content out to plain text.

Alternatively, if `newglossaryentry` allows nested tex for name/description, you could run `myst-to-tex` on the children... I suspect that's overkill though, and `toText` is best...?